File: branches/Medusa/Source/Fantasy.Base/XSerialization/XObjectConverter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.ComponentModel;

namespace Fantasy.XSerialization
{
    [XConverter(TargetType=typeof(Object))]
    public class XObjectConverter : TypeConverter
    {
        public override object ConvertTo(ITypeDescriptorContext context, System.Globalization.CultureInfo culture, object value, Type destinationType)
        {
            if (value != null)
            {
                return Convert.ChangeType(value, destinationType);
            }
            else
            {
                return null;
            }
        }

        public override object ConvertFrom(ITypeDescriptorContext context, System.Globalization.CultureInfo culture, object value)
        {
            if (value == null)
            {
                return null;
            }
            else
            {
                return Convert.ChangeType(value, typeof(string));
            }
        }
    }
}

File: AutoConverted/Fantasy.Base/XSerialization/XObjectConverter.java
package Fantasy.XSerialization;

//C# TO JAVA CONVERTER TODO TASK: Java annotations will not correspond to .NET attributes:
//[XConverter(TargetType=typeof(Object))]
public class XObjectConverter extends TypeConverter
{
	@Override
	public Object ConvertTo(ITypeDescriptorContext context, System.Globalization.CultureInfo culture, Object value, java.lang.Class destinationType)
	{
		if (value != null)
		{
			return Convert.ChangeType(value, destinationType);
		}
		else
		{
			return null;
		}
	}

	@Override
	public Object ConvertFrom(ITypeDescriptorContext context, System.Globalization.CultureInfo culture, Object value)
	{
		if (value == null)
		{
			return null;
		}
		else
		{
			return Convert.ChangeType(value, String.class);
		}
	}
}